rest/data/test_results.go

```diff
@@ -56,81 +73,131 @@ func (dbc *DBConnector) FindTestResultByTestName(ctx context.Context, opts TestR
+	if err != nil {
+		return nil, err
+	}
+
+	var combinedResults []dbModel.TestResult
+	for i := range testResultsDocs {
+		results, err := testResultsDocs[i].Download(ctx)
+		if err != nil {
+			return nil, err
+		}
+		combinedResults = append(combinedResults, results...)
+	}
+
+	return importTestResults(ctx, combinedResults)
+}
+
+func (mc *MockConnector) findTestResultsByTaskID(ctx context.Context, opts TestResultsOptions) (*dbModel.TestResults, error) {
+	var testResultsDoc *dbModel.TestResults
 
 	for key, _ := range mc.CachedTestResults {
 		tr := mc.CachedTestResults[key]
 		if opts.EmptyExecution {
-			if tr.Info.TaskID == opts.TaskID && (testResults == nil || tr.Info.Execution > testResults.Info.Execution) {
-				testResults = &tr
+			if tr.Info.TaskID == opts.TaskID && (testResultsDoc == nil || tr.Info.Execution > testResultsDoc.Info.Execution) {
+				testResultsDoc = &tr
 			}
 		} else {
 			if tr.Info.TaskID == opts.TaskID && tr.Info.Execution == opts.Execution {
-				testResults = &tr
+				testResultsDoc = &tr
 				break
 			}
 		}
 	}
 
-	if testResults == nil {
+	if testResultsDoc == nil {
 		return nil, gimlet.ErrorResponse{
 			StatusCode: http.StatusNotFound,
 			Message:    "test results not found",
 		}
 	}
 
-	results, err := testResults.Download(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	return importTestResults(ctx, results)
+	return testResultsDoc, nil
 }
 
-func (mc *MockConnector) findTestResultsByDisplayTaskID(ctx context.Context, opts TestResultsOptions) ([]model.APITestResult, error) {
+func (mc *MockConnector) findTestResultsByDisplayTaskID(ctx context.Context, opts TestResultsOptions) ([]dbModel.TestResults, error) {
 	var (
-		testResults     []*dbModel.TestResults
-		combinedResults []dbModel.TestResult
+		testResultsDocs []dbModel.TestResults
 		latestExecution int
 	)
 
 	for key, _ := range mc.CachedTestResults {
 		tr := mc.CachedTestResults[key]
 		if tr.Info.DisplayTaskID == opts.DisplayTaskID {
 			if opts.EmptyExecution || tr.Info.Execution == opts.Execution {
-				testResults = append(testResults, &tr)
+				testResultsDocs = append(testResultsDocs, tr)
 			}
 		}
 		if tr.Info.Execution > latestExecution {
 			latestExecution = tr.Info.Execution
 		}
 	}
 
-	if testResults == nil {
-		return nil, gimlet.ErrorResponse{
-			StatusCode: http.StatusNotFound,
-			Message:    "test results not found",
+	if opts.EmptyExecution {
+		var filteredDocs []dbModel.TestResults
+		for i := range testResultsDocs {
+			if testResultsDocs[i].Info.Execution == latestExecution {
+				filteredDocs = append(filteredDocs, testResultsDocs[i])
+			}
 		}
+		testResultsDocs = filteredDocs
 	}
 
-	for i := range testResults {
-		if !opts.EmptyExecution || testResults[i].Info.Execution == latestExecution {
-			results, err := testResults[i].Download(ctx)
-			if err != nil {
-				return nil, err
-			}
-			combinedResults = append(combinedResults, results...)
+	if testResultsDocs == nil {
```

Comment:
Should this check if `len(testResultsDocs) == 0` to cover both nil and empty slice cases?